### PR TITLE
Dartls server config windows

### PR DESCRIPTION
**Pull request by erietz-werner**
I was having a terrible time getting the client to attach to the buffer on Windows. This fixes the issue for me.

### CODE REVIEW

File: lua/lspconfig/server_configurations/dartls.lua [hunk 1]
```diff
@@ -1,8 +1,15 @@
 local util = require 'lspconfig.util'
 
+local bin_name = 'dart'
+local cmd = { bin_name, 'language-server', '--protocol=lsp' }
+
+if vim.fn.has 'win32' == 1 then
+  cmd = { 'cmd.exe', '/C', bin_name, 'language-server', '--protocol=lsp' }
+end
```
Comment (justinmk) [inline lua/lspconfig/server_configurations/dartls.lua:8]: ternary is more readable, avoids nesting, etc.

```suggestion
local cmd = (vim.fn.has 'win32' == 1
    and { 'cmd.exe', '/C', bin_name, 'language-server', '--protocol=lsp' }
    or { bin_name, 'language-server', '--protocol=lsp' })
```

Comment (erietz-werner) [reply lua/lspconfig/server_configurations/dartls.lua]: Thanks, I didn't know about the lua ternary.

**Comment by justinmk** [comment]
sure seems like vim.lsp or lspconfig should be doing this by default. I keep seeing PRs like this.

**Comment by erietz-werner** [comment]
Yes, it is in a ton of the server configurations. That is where I got the idea from. `grep -R "vim.fn.has 'win32'" | wc -l`

**Pull request merged**